### ISSUE TITLE
force-close without broadcast

### ISSUE THREAD

**Issue by ecdsa** [ACINQ/eclair#2358]
See full report here: https://github.com/spesmilo/electrum/issues/7907

The channel was in the state DATA_WAIT_FOR_FUNDING_LOCKED.

Electrum requested a force-close from a backup, using ChannelReestablish with an old state.
Electrum did not send an error message (the user was using a version that is older than this commit https://github.com/spesmilo/electrum/commit/90dbac5a65a8b7804a5b02a6cf598588a06a71a3 ), so Eclair did not force-close (that is expected).

I then tried to force-close manually for this user, using the CLI:
`eclair forceclose --channelId="30f63cdfd5c47f4fe4d5a5284e99297fc56ff699767bc45cca781dbf52a89ffb"`

The CLI response was that the channel was being closed, but Eclair did not broadcast a commitment transaction.
Here is the relevant part of the log :
```
2022-07-28 17:29:31,924 INFO  f.a.eclair.Diagnostics n:02eec015e88bf306b31fe2cb4907c8730aa609b52b52946c82fc5bc8af9978ffb1 c:30f63cdfd5c47f4fe4d5a5284e99297fc56ff699767bc45cca781dbf52a89ffb - OUT msg=FundingLocked(30f63cdfd5c47f4fe4d5a5284e99297fc56ff699767bc45cca781dbf52a89ffb,02f0d18b74070538f0f0e0ab1b118874cb294813d6d5c169eb8f99531d6251e5a8,TlvStream(List(),List()))
2022-07-28 18:36:13,181 WARN  f.a.e.channel.Channel n:02eec015e88bf306b31fe2cb4907c8730aa609b52b52946c82fc5bc8af9978ffb1 c:30f63cdfd5c47f4fe4d5a5284e99297fc56ff699767bc45cca781dbf52a89ffb - force-closing channel at user request
2022-07-28 18:36:23,196 INFO  f.a.e.channel.Channel n:02eec015e88bf306b31fe2cb4907c8730aa609b52b52946c82fc5bc8af9978ffb1 c:30f63cdfd5c47f4fe4d5a5284e99297fc56ff699767bc45cca781dbf52a89ffb - deleting database record for channelId=30f63cdfd5c47f4fe4d5a5284e99297fc56ff699767bc45cca781dbf52a89ffb
2022-07-28 18:36:23,197 INFO  f.a.e.channel.Channel n:02eec015e88bf306b31fe2cb4907c8730aa609b52b52946c82fc5bc8af9978ffb1 c:30f63cdfd5c47f4fe4d5a5284e99297fc56ff699767bc45cca781dbf52a89ffb - shutting down
2022-07-28 19:05:35,381 INFO  f.a.eclair.Diagnostics CON n:02eec015e88bf306b31fe2cb4907c8730aa609b52b52946c82fc5bc8af9978ffb1 c:30f63cdfd5c47f4fe4d5a5284e99297fc56ff699767bc45cca781dbf52a89ffb - IN msg=ChannelReestablish(30f63cdfd5c47f4fe4d5a5284e99297fc56ff699767bc45cca781dbf52a89ffb,0,0,PrivateKey(0000000000000000000000000000000000000000000000000000000000000000),02fe8d1eb1bcb3432b1db5833ff5f2226d9cb5e65cee430558c18ed3a3c86ce1af,TlvStream(List(),List()))

```

I was able to recreate the commitment transaction using scripts, because I had a backup of the channels database.
However, without that backup funds would have been lost.

eclair version: eclair-node-0.7.0-a804905

**Comment by t-bast** [comment]
This is a duplicate of #2021, isn't it? The eclair node has nothing at stake (no output on the commit tx and channel was never used), so we fast-track the closing process.

**Comment by ecdsa** [comment]
Not sure if this is a duplicate. #1730 seems to have been fixed.
Eclair 0.7 does broadcast the commitment tx in the conditions described in #1730, even with nothing at stake. 
What is new (at least to me) is that the issue is still there if Eclair never received funding_locked.

**Comment by ecdsa** [comment]
sorry, my bad. it is indeed a duplicate, #1730 has not been fixed.

**Comment by t-bast** [comment]
Sorry we let #1730 slip for too long, I created a PR to fix it (#2360).
We had actually already fixed most occurrences of this issue, but there were a few edge cases remaining.
